scripts/gcpkms-encrypt.sh

```diff
@@ -0,0 +1,58 @@
+#! /usr/bin/env bash
+set -xe
+set -o pipefail
+
+#
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# This script is designed to be called from a terraform `external` data source.
+# Arguments are passed as JSON and evaluated as bash variables.
+# The output format is JSON.
+#
+
+
+# Extract args into shell variables
+eval "$(jq -r '@sh "ROOT=\(.root) DATA=\(.data) PROJECT=\(.project) LOCATION=\(.location) KEYRING=\(.keyring) KEY=\(.key)"')"
+
+# If no root or data was given, error
+[[ "${ROOT}" == "null" ]] && echo "Missing ROOT!" && exit 1
+[[ "${DATA}" == "null" ]] && echo "Missing DATA!" && exit 1
+
+# Calculate the md5 of the data - used to lookup
+SIG=$(echo -n "${DATA}|${PROJECT}|${LOCATION}|${KEYRING}|${KEY}" | md5 | cut -d ' ' -f1)
```

Comment:
This did not work in cloud shell.

Here is a proposed cross-platform fix:

```bash
# Calculate the md5 of the data - used to lookup
MD5SUM=$(command -v md5 || true)
[[ -z "${MD5SUM}" ]] && MD5SUM=$(command -v md5sum || true)
[[ -z "${MD5SUM}" ]] && echo "Missing command: 'md5' or 'md5sum'" >&2 && exit 1

SIG=$(echo -n "${DATA}|${PROJECT}|${LOCATION}|${KEYRING}|${KEY}" | ${MD5SUM} | cut -d ' ' -f1)
```

Reply:
Turns out md5sum actually works everywhere and md5 is just an alias on darwin. Updated.